naming-conventions.md

```diff
@@ -226,6 +226,53 @@ This will produce the following JSON:
 }
 ```
 
+### Request/Response types
+
+For complicated `Handler` requests or responses, where we use a custom type
+instead of an `Entity` directly, such types should be named `{Resource}{Method}`
+where `Resource` matches the route and `Method` is Sentence-case.
+
+```hs
+-- Good
+data SchoolGet
+  -- ...
+
+instance ToJSON SchoolGet where
+  -- ...
+
+getSchoolR :: SchoolId -> Handler Value
+getSchoolR schoolId = do
+  school <- get404 schoolId
+  sendStatusJSON status200 SchoolGet {- ... -}
+
+data SchoolsPost
+  -- ...
+
+instance FromJSON SchoolsPost where
+  -- ...
+
+postSchoolsR :: Handler Value
+postSchoolsR = do
+  SchoolsPost {..} <- requireJsonBody
+  -- ...
```

Comment:
Odd (for us) case - if you have a response for a `POST`, do you do `SchoolsPostRequest` and `SchoolsPostResponse`?

Reply:
Yeah, I thought of that and purposely didn't talk about. Partly because we have no cases (that I know of) and partly because I'm not sure what to prescribe. If you'd like to see a sentence in here about that (either prescribing something or just saying that it's left unspecified) please let me know.

Reply:
Eh, probably better to leave it unspecified. I suspect if we do ever have `POST`s that return something, the result will be an `Entity` potentially `WithMetadata`